c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -173,6 +173,16 @@ impl<'tcx> TypeChecker<'tcx, '_> {
                             self.do_assign(elem_lty, op_lty);
                         }
                     }
+                    AggregateKind::Adt(adt_did, ..) => {
+                        let base_adt_def = self.acx.tcx().adt_def(adt_did);
+                        let fields = &base_adt_def.non_enum_variant().fields;
+                        for (field, op) in fields.iter().zip(ops.iter()) {
+                            let op_lty = self.acx.type_of(op);
+                            let field_lty = self.acx.gacx.field_tys[&field.did];
+                            // Pseudo-assign from each operand to the element type of the field.
+                            self.do_assign(field_lty, op_lty);
```

Comment:
We know the unsubstituted `field_lty` is the wrong thing to use here in general.  Please at least add a TODO comment (or better yet, a relevant assert), or else put the `subst` call back in (it's fine if it panics when `args.len() > 0` for now, until we fix `subst` properly).

Reply:
renamed in 7de4be76